### PR TITLE
Vendor k/*@v0.23 and controller-runtime@v0.11

### PR DESCRIPTION
**Pull request by acumino**
**How to categorize this PR?**

/area open-source
/kind task

**What this PR does / why we need it**:
Vendors k8s.io/*@v0.23.3, sigs.k8s.io/controller-runtime@v0.11.0 and sigs.k8s.io/controller-tools@v0.8.0 and adapts to the breaking / most notable changes

**Which issue(s) this PR fixes**:
Part of #5282

**Special notes for your reviewer**:

**Release note**:

```other operator
The following golang dependencies have been upgraded, please consult the upstream release notes and [this issue](https://github.com/gardener/gardener/issues/5282) for guidance on upgrading your golang dependencies when vendoring this gardener version:
- `k8s.io/*` to `v0.23.3`
- `sigs.k8s.io/controller-runtime` to `v0.11.0`
- `sigs.k8s.io/controller-tools` to `v0.8.0`
```

### CODE REVIEW

**Review by rfranzke — COMMENTED** [review]
Thanks for the PR and thanks to @timebertt helping with structuring it :)

Can you explain why this is only "Part of https://github.com/gardener/gardener/issues/5282"? What is missing and can't we do this already now to get this topic closed?

**Comment by acumino** [comment]
> Can you explain why this is only "Part of #5282"? What is missing and can't we do this already now to get this topic closed?

There are some enhancements that can be done ex - `consider replacing custom coding in seed-admission-controller with builder.WebhookBuilder.{WithDefaulter,WithValidator}`
and also `Bumping of envtest binaries`

**Comment by rfranzke** [comment]
Thanks! What is preventing us from doing this now already as part of this PR?

**Comment by acumino** [comment]
> Thanks! What is preventing us from doing this now already as part of this PR?

This PR only covers the breaking changes. Other changes are not important so can be done in separate PR, so this PR will be easy to follow. 
I can also make those changes here if you suggest so.

**Comment by rfranzke** [comment]
> I can also make those changes here if you suggest so.

You decide. The only thing I would like to ask for is that we actively pursue the whatever other changes we plan before https://github.com/gardener/gardener/issues/5282 can be closed. I would like to prevent the situation which happened to #4772 (open since long).

**Comment by acumino** [comment]
> > I can also make those changes here if you suggest so.
> 
> You decide. The only thing I would like to ask for is that we actively pursue the whatever other changes we plan before #5282 can be closed. I would like to prevent the situation which happened to #4772 (open since long).

I prefer doing it separately that seems more reasonable to me as K/K kind of follow the same pattern, cause envtest is part of a different module. I will resolve all other things once this PR is merged.

**Review by rfranzke — DISMISSED** [review]
Thanks @acumino and @timebertt !

/lgtm

File: go.mod [hunk 1]
```diff
@@ -19,11 +19,11 @@ require (
 	github.com/gardener/landscaper/apis v0.7.0
 	github.com/gardener/machine-controller-manager v0.41.0
 	github.com/ghodss/yaml v1.0.0
-	github.com/go-logr/logr v0.4.0
+	github.com/go-logr/logr v1.2.0
```
Comment (timebertt) [inline go.mod:22]: Directly go for `v1.2.2`?

File: go.mod [hunk 2]
```diff
@@ -79,18 +79,18 @@ require (
 replace (
 	github.com/emicklei/go-restful => github.com/emicklei/go-restful v2.9.5+incompatible // keep this value in sync with k8s.io/apiserver
 	github.com/envoyproxy/go-control-plane => github.com/envoyproxy/go-control-plane v0.9.4
-	github.com/googleapis/gnostic => github.com/googleapis/gnostic v0.5.5 // keep this value in sync with k8s.io/apiserver
+	github.com/google/gnostic => github.com/google/gnostic v0.6.6 // keep this value in sync with k8s.io/apiserver
```
Comment (timebertt) [inline go.mod]: This value is not in sync with https://github.com/kubernetes/apiserver/blob/7d5ba2d02807fe7c7c7dd8b43b51bd708a3ee35b/go.mod#L21
Why did you bump it?

File: go.mod [hunk 3]
```diff
@@ -81,7 +81,7 @@ replace (
 	github.com/emicklei/go-restful => github.com/emicklei/go-restful v2.9.5+incompatible // keep this value in sync with k8s.io/apiserver
 	github.com/envoyproxy/go-control-plane => github.com/envoyproxy/go-control-plane v0.9.4
 	github.com/google/gnostic => github.com/google/gnostic v0.6.6 // keep this value in sync with k8s.io/apiserver
-	github.com/prometheus/client_golang => github.com/prometheus/client_golang v1.11.0 // keep this value in sync with sigs.k8s.io/controller-runtime
+	github.com/prometheus/client_golang => github.com/prometheus/client_golang v1.12.1 // keep this value in sync with sigs.k8s.io/controller-runtime
```
Comment (timebertt) [inline go.mod]: This value is not in sync with https://github.com/kubernetes-sigs/controller-runtime/blob/f236f0345ad2933912ebf34bfcf0f93620769654/go.mod#L12
Why did you bump it?

File: go.mod [hunk 4]
```diff
@@ -71,8 +71,8 @@ require (
 	k8s.io/kubelet v0.23.3
 	k8s.io/metrics v0.23.3
 	k8s.io/utils v0.0.0-20211116205334-6203023598ed
-	sigs.k8s.io/controller-runtime v0.10.2
-	sigs.k8s.io/controller-runtime/tools/setup-envtest v0.0.0-20211005045149-78ce10e2ebad // v0.10.2
+	sigs.k8s.io/controller-runtime v0.11.0
+	sigs.k8s.io/controller-runtime/tools/setup-envtest 2df0a5baf13e9b882a2868acbe56f4f43801d98c // v0.11.0
```
Comment (timebertt) [inline go.mod]: This commit is not tagged with `v0.11.0`. Shouldn't it rather be https://github.com/kubernetes-sigs/controller-runtime/commit/f236f0345ad2933912ebf34bfcf0f93620769654?

**Review by timebertt — CHANGES_REQUESTED** [review]
Very nice PR, @acumino. It was very easy to follow and review. Well done!
I have a few suggestions regarding the concrete versions, but overall this looks very good :)

File: go.mod (hunk 1 above)
Comment (acumino) [reply go.mod:22]: I didn't update, it was brought by k8s. Should I bump it to 1.2.2?

File: go.mod (hunk 2 above)
Comment (acumino) [reply go.mod]: For the latest version, I will make it in sync.

File: go.mod (hunk 1 above)
Comment (timebertt) [reply go.mod:22]: Fine, let's keep it for now.

**Review by timebertt — APPROVED** [review]
Thanks again!
/lgtm

**Pull request merged**